tests/test_preprocessing/test_columnselector.py

```diff
@@ -0,0 +1,59 @@
+import pandas as pd
+from pandas.testing import assert_frame_equal
+from sklearn.pipeline import make_pipeline
+import pytest
+from sklego.preprocessing import ColumnSelector
+
+
+@pytest.fixture()
+def df():
+    return pd.DataFrame({"a": [1, 2, 3, 4, 5, 6],
+                         "b": [10, 9, 8, 7, 6, 5],
+                         "c": ["a", "b", "a", "b", "c", "c"],
+                         "d": ["b", "a", "a", "b", "a", "b"],
+                         "e": [0, 1, 0, 1, 0, 1]})
+
+
+def test_select_two(df):
+    result_df = ColumnSelector(['d', 'e']).fit_transform(df)
+    expected_df = pd.DataFrame({
+        "d": ["b", "a", "a", "b", "a", "b"],
+        "e": [0, 1, 0, 1, 0, 1]})
+
+    assert_frame_equal(result_df, expected_df)
+
+
+def test_select_one(df):
+    result_df = ColumnSelector(['e']).fit_transform(df)
+    expected_df = pd.DataFrame({
+        "e": [0, 1, 0, 1, 0, 1]})
+
+    assert_frame_equal(result_df, expected_df)
+
+
+def test_select_none(df):
+    result_df = ColumnSelector([]).fit_transform(df)
```

Comment:
Shouldnt this just be a `ValueError`? I can't imagine a use-case atm and the next step in the pipeline will give an error, while the cause is in this step.

Reply:
In that case we need to throw the same error in ColumnDroper if someone drops all columns from the dataframe.
Either throw `ValueError` in both cases or in none.

Reply:
@MBrouns should I throw VlaueError in both cases?

Reply:
I would say to add something along these lines to fit for the columnselector:

```python
if len(self.columns) == 0:
    raise ValueError("expected `columns` to be at least of length 1, found length of 0 instead")
```

and for the columndropper:

```python
if len(self.feature_names_) == 0:
   raise ValueError(f"using the ColumnDropper for dropping {self.columns} would result in an empty output dataset ")
```

Reply:
@MBrouns I added this check undef _check_column_length(), please review

Reply:
Please review this so that I can move ahead in this PR if in case anything else is to be done

Reply:
Don't worry, we will have a look but it is fair to point out that we're all volunteers here and haste makes waste. Also ... it's dinnertime here and I will cook for my household in a moment so this might need to wait until later.

Reply:
I will surely take care of that from now :)